File: src/main/java/alfheim/common/network/Message1d.java
package alfheim.common.network;

import alexsocol.asjlib.network.ASJPacket;

public class Message1d extends ASJPacket {

	public int type;
	public double data1;
	
	public Message1d(m1d ty, double d1) {
		type = ty.ordinal();
		data1 = d1;
	}
	
	public enum m1d {
		DEATH_TIMER, ELVEN_FLIGHT_MAX, ESMABIL, GINNUNGAGAP, KNOWLEDGE, TIME_STOP_REMOVE
	}
}

File: src/main/java/alfheim/common/network/Message1d.kt
package alfheim.common.network

import alexsocol.asjlib.network.ASJPacket
import alfheim.client.core.handler.PacketHandlerClient
import cpw.mods.fml.common.network.simpleimpl.*

class Message1d(ty: m1d, val data1: Double): ASJPacket() {
	
	val type: Int
	
	init {
		type = ty.ordinal
	}
	
	enum class m1d {
		DEATH_TIMER, TIME_STOP_REMOVE, KNOWLEDGE, CL_SLOWDOWN
	}
	
	class Handler: IMessageHandler<Message1d, IMessage> {
		
		override fun onMessage(packet: Message1d, message: MessageContext): IMessage? {
			PacketHandlerClient.handle(packet)
			return null
		}
	}
}
